marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java

```diff
@@ -83,12 +92,53 @@ private int getBatchSize() {
 
 		@Override
 		public void accept(InputStream input) {
-			// TODO Auto-generated method stub
+			
+			InputStream output = null;
+			try {
+				queue.put(input);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+			if((queue.size()% getBatchSize()) > 0)
+				return;
+			
+			switch (getPhase()) {
+				case INITIALIZING:
+					setPhase(WorkPhase.RUNNING);
+					break;
+				case RUNNING:
+					break;
+				case INTERRUPTING:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				case INTERRUPTED:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				case COMPLETED:
```

Comment:
Do these cases need separate throw actions, given that cases fall through in Java?

Also, is it accurate to report the problem as an internal exception?  Would a Java IllegalStateException be more accurate?